src/components/activity/ActivityParticipantCollectionModal.tsx

```diff
@@ -146,78 +146,102 @@ const ActivityParticipantCollectionModal: React.FC<
     </AdminModal>
   )
 }
+const rawActivitySessionResDataSchema = yup.array().of(
+  yup.object({
+    id: yup.string().required(),
+    title: yup.string().nullable(),
+    participants: yup.array().of(
+      yup.object({
+        id: yup.string().required(),
+        name: yup.string().nullable(),
+        phone: yup.string().nullable(),
+        email: yup.string().nullable(),
+        orderLogId: yup.string().required(),
+        attended: yup.boolean().nullable(), // If the value is null, it indicates non-attendance
+        activityTicketTitle: yup.string().nullable(),
+      }),
+    ),
+  }),
+)
 
 const useActivitySessionParticipants = (activityId: string) => {
-  const { loading, error, data, refetch } = useQuery<
-    hasura.GET_ACTIVITY_PARTICIPANTS,
-    hasura.GET_ACTIVITY_PARTICIPANTSVariables
-  >(
-    gql`
-      query GET_ACTIVITY_PARTICIPANTS($activityId: uuid!) {
-        activity_enrollment(where: { activity_id: { _eq: $activityId } }, order_by: {}) {
-          activity_session_id
-          order_log_id
-          member_id
-          member_name
-          member_email
-          member_phone
-          order_log_id
-          attended
-          activity_ticket {
-            id
-            title
+  const [loading, setLoading] = useState<boolean>(true)
+  const [error, setError] = useState<string | null>(null)
+  const [sessions, setSessions] = useState<ActivitySessionParticipantsDTO[]>([])
+  const { authToken } = useAuth()
+
+  const fetchActivitySessionParticipants = useCallback(
+    async (activityId: string) => {
+      if (authToken && activityId) {
+        try {
+          setLoading(true)
+          const response = await axios.get(
+            `${process.env.REACT_APP_LODESTAR_SERVER_ENDPOINT}/activity/${activityId}/participants`,
+            {
+              headers: {
+                Authorization: `Bearer ${authToken}`,
+              },
+            },
+          )
+          const { activitySessions: data } = response.data
+
+          try {
+            await rawActivitySessionResDataSchema.validate(data)
+          } catch (error) {
+            const errorMessage = (error as Error).message || 'An unknown error occurred'
+            console.error(activityId, errorMessage)
+            setError(errorMessage)
+            return
           }
-        }
-        activity_session(where: { activity_id: { _eq: $activityId } }, order_by: { started_at: asc }) {
-          id
-          title
-          started_at
+
+          const mapApiDataToDTO = (data: ActivitySessionParticipantResDto[]): ActivitySessionParticipantsDTO[] => {
+            const idCounts: { [key: string]: number } = {}
+            return data.map((sessionData: any) => ({
```

Comment:
prefer defined type

Reply:
I removed the type here since I already defined it in line #197